src/main/java/org/jpeek/calculus/java/Ccm.java

```diff
@@ -52,57 +61,163 @@ public XML node(
                 ).toString()
             );
         }
-        return Ccm.withFixedNcc(
-            new XSLDocument(
-                new UncheckedInput(
-                    new ResourceOf("org/jpeek/metrics/CCM.xsl")
-                ).stream()
-            ).transform(skeleton),
-            skeleton
+        final XSLDocument doc = new XSLDocument(
+            new UncheckedInput(
+                new ResourceOf("org/jpeek/metrics/CCM.xsl")
+            ).stream()
         );
+        final XML meta = addMetaInformation(skeleton, params);
+        return doc.transform(meta);
     }
 
     /**
-     * Updates the transformed xml with proper NCC value.
-     * @param transformed The transformed XML skeleton.
-     * @param skeleton XML Skeleton
-     * @return XML with fixed NCC.
+     * Adds meta information to the skeleton XML document.
+     * This method modifies the skeleton XML document by adding meta information
+     * about the computed CCM metric.
+     * @param skeleton The skeleton XML document representing the code structure.
+     * @param params Parameters for the computation.
+     * @return The modified XML document containing meta information.
      */
-    private static XML withFixedNcc(final XML transformed, final XML skeleton) {
-        final List<XML> packages = transformed.nodes("//package");
-        for (final XML elt : packages) {
-            final String pack = elt.xpath("/@id").get(0);
-            final List<XML> classes = elt.nodes("//class");
-            for (final XML clazz : classes) {
-                Ccm.updateNcc(skeleton, pack, clazz);
+    private static XML addMetaInformation(final XML skeleton, final Map<String, Object> params) {
+        try {
+            final Document doc = DocumentBuilderFactory.newInstance().newDocumentBuilder()
+                .newDocument();
+            final Element meta = doc.createElement("meta");
```

Comment:
it is not required to change XML document and add some data, it is easier to call XSL with parameter, so you need to read ncc value in XSL and use it.

Reply:
You do not understand, we can run XSL for class skeleton with parameter cmm, so we do not need to create package and all others inside, as all this arledy implemented and can be changed in case if it is needed in xsl using common way. it is absolutely not readable in code, xsl is much clear